pandas/io/excel/_base.py

```diff
@@ -350,6 +350,8 @@ def __init__(self, filepath_or_buffer):
             self.book = self.load_workbook(filepath_or_buffer)
         elif isinstance(filepath_or_buffer, str):
             self.book = self.load_workbook(filepath_or_buffer)
+        elif isinstance(filepath_or_buffer, bytes):
+            self.book = self.load_workbook(BytesIO(filepath_or_buffer))
```

Comment:
Is BytesIO absolutely required here? Seems like we could treat str / bytes more consistently

Reply:
> Seems like we could treat str / bytes more consistently

@WillAyd : Consistently how?

Reply:
We aren't wrapping str with an IO object so asking if that's really required for bytes

Reply:
Gotcha.  Thanks for clarifying!

@ftruzzi : Can you confirm if `BytesIO` is needed or not?

I suspect the answer is yes, but makes sense to double check for future reference.

Reply:
Yes I think it is needed. Probably `str` works just for filenames and not for stringified objects?

I had tried to treat `str` and `buffer` the same way, which could work with xlrd by passing the file contents to the `file_contents` kwarg in `xlrd.open_workbook` but I found no way to do the same thing in openpyxl and other engines.

And just by skipping `BytesIO`, engines complain as they try to open the thing as if it was a file:
```
File "C:\Users\ftruzzi\Documents\projects\bixby\capsule-dev-it\venv\lib\site-packages\xlrd\__init__.py", line 111, in open_workbook
    with open(filename, "rb") as f:
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x91 in position 14: invalid start byte
```
and
```
openpyxl.utils.exceptions.InvalidFileException: openpyxl does not support b'.relspk\x05\x06\x00\x00\x00\x00\x1d\x00\x1d\x00\xe1\x07\x00\x00\x8f\xa9\x0e\x00\x00\x00' file format, please check you can open it with Excel first. Supported formats are: .xlsx,.xlsm,.xltx,.xltm
```

Reply:
Yes I think it is needed. Probably `str` works just for filenames and not for stringified objects?

I had tried to treat `str` and `buffer` the same way, which could work with xlrd by passing the file contents to the `file_contents` kwarg in `xlrd.open_workbook` but I found no way to do the same thing in openpyxl and other engines.

And just by skipping `BytesIO`, engines complain as they try to open the thing as if it was a file:
```
File "...\xlrd\__init__.py", line 111, in open_workbook
    with open(filename, "rb") as f:
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x91 in position 14: invalid start byte
```
and
```
openpyxl.utils.exceptions.InvalidFileException: openpyxl does not support b'.relspk\x05\x06\x00\x00\x00\x00\x1d\x00\x1d\x00\xe1\x07\x00\x00\x8f\xa9\x0e\x00\x00\x00' file format, please check you can open it with Excel first. Supported formats are: .xlsx,.xlsm,.xltx,.xltm
```